tuya.cpp

```diff
@@ -513,6 +514,44 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
                         }
                     }
                     break;
+                    
+                    // Dimmer level for mode 1
+                    case 0x0202:
+                    {
+                        if (R_GetProductId(lightNode) == QLatin1String("Tuya_DIMSWITCH Earda Dimmer") ||
+                            R_GetProductId(lightNode) == QLatin1String("Tuya_DIMSWITCH EDM-1ZAA-EU"))
+                        {
+                            quint8 bri = static_cast<quint8>(data * 254 / 1000); // 0 to 1000 value
+                            
+                            ResourceItem *item = lightNode->item(RStateBri);
+                            if (item && item->toNumber() != bri)
+                            {
+                                item->setValue(bri);
+                                Event e(RLights, RStateBri, lightNode->id(), item);
+                                enqueueEvent(e);
+                                update = true;
+                            }
+                        }
+                    }
+                    break;
+                    // Dimmer level for mode 2
+                    case 0x0203:
+                    {
+                        if (R_GetProductId(lightNode) == QLatin1String("Tuya_DIMSWITCH Not model found yet"))
+                        {
+                            quint8 bri = static_cast<quint8>(data * 254 / 1000); // 0 to 1000 value
```

Comment:
`const qint64 bri = data * 254 / 1000; // 0 to 1000 value`

Same, don't need to cast here since `ResourceItem::setValue(qint64 val)` takes an qint64 anyway.

Reply:
done